From: Aaron Tomlin <atomlin@redhat.com>
Date: Mon, 23 Jun 2014 13:22:05 -0700
Subject: nmi: provide the option to issue an NMI back trace to every cpu but
 current
Patch-mainline: Never, SUSE-Xen specific
References: bsc#940946, bsc#937444

Sometimes it is preferred not to use the trigger_all_cpu_backtrace()
routine when one wants to avoid capturing a back trace for current.  For
instance if one was previously captured recently.

This patch provides a new routine namely
trigger_allbutself_cpu_backtrace() which offers the flexibility to issue
an NMI to every cpu but current and capture a back trace accordingly.

Patch x86 and sparc to support new routine.

[dzickus@redhat.com: add stub in #else clause]
[dzickus@redhat.com: don't print message in single processor case, wrap with get/put_cpu based on Oleg's suggestion]
[sfr@canb.auug.org.au: undo C99ism]
Signed-off-by: Aaron Tomlin <atomlin@redhat.com>
Signed-off-by: Don Zickus <dzickus@redhat.com>
Acked-by: David S. Miller <davem@davemloft.net>
Cc: Mateusz Guzik <mguzik@redhat.com>
Cc: Oleg Nesterov <oleg@redhat.com>
Signed-off-by: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Acked-by: Borislav Petkov <bp@suse.de>

Automatically created from "patches.fixes/02-nmi-provide-the-option-to-issue-an-nmi-back-trace-to-every-cpu-but-current.patch" by xen-port-patches.py

--- a/arch/x86/kernel/apic/hw_nmi.c
+++ b/arch/x86/kernel/apic/hw_nmi.c
@@ -61,7 +61,7 @@ void arch_trigger_all_cpu_backtrace(bool
 #ifndef CONFIG_XEN
 		apic->send_IPI_mask(to_cpumask(backtrace_mask), NMI_VECTOR);
 #else /* this works even without CONFIG_X86_LOCAL_APIC */
-	xen_send_IPI_all(NMI_VECTOR);
+		xen_send_IPI_mask(to_cpumask(backtrace_mask), NMI_VECTOR);
 #endif
 	}
 
